Content.Server/Sound/EmitSoundSystem.cs

```diff
@@ -32,6 +32,25 @@ public override void Update(float frameTime)
                 TryEmitSound(uid, soundSpammer);
             }
         }
+
+        var intervalQuery = EntityQueryEnumerator<EmitSoundIntervalComponent>();
+
+        while (intervalQuery.MoveNext(out var uid, out var comp))
+        {
+            if (!comp.Enabled)
+            {
+                // While disabled, keep pushing the next emit time out,
+                // effectively pausing the timer.
+                comp.NextEmitTime += TimeSpan.FromSeconds(frameTime);
+                continue;
+            }
+
+            if (Timing.CurTime > comp.NextEmitTime)
+            {
+                TryEmitSound(uid, comp);
+                SelectNextInterval(uid, comp);
+            }
+        }
     }
```

Comment:
Is this not just a copy of spamemitsound.

Reply:
I want a randomized interval between sounds, rather than the fixed one SpamEmitSound uses. I probably could expand SpamEmitSound a little to do what I need. An optional MaxExtraInterval property and a float to store the selected extra interval each iteration should do it. Sounds good.